wyvern-exchange/src/wyvern-exchange.ts

```diff
@@ -0,0 +1,283 @@
+    call.inputs.uints[9], //makerRelayerFee:
+    call.inputs.uints[10], // takerRelayerFee:
+    call.inputs.uints[11], //makerProtocolFee:
+    call.inputs.uints[12], //takerProtocolFee:
+    call.inputs.addrs[10].toHexString(), // feeRecipient:
+    orders.helpers.getFeeMethod(call.inputs.feeMethodsSidesKindsHowToCalls[4]), //feeMethod:
+    orders.helpers.getOrderSide(call.inputs.feeMethodsSidesKindsHowToCalls[5]), //side:
+    orders.helpers.getSaleKind(call.inputs.feeMethodsSidesKindsHowToCalls[6]), //saleKind:
+    call.inputs.addrs[11].toHexString(), //target:
+    orders.helpers.getHowToCall(call.inputs.feeMethodsSidesKindsHowToCalls[7]), //howToCall:
+    call.inputs.calldataSell, // callData:
+    call.inputs.replacementPatternSell, //replacementPattern:
+    call.inputs.addrs[12], //staticTarget:
+    call.inputs.staticExtradataSell, //staticExtradata:
+    paymentToken.toHexString(), //paymentToken:
+    call.inputs.uints[13], //basePrice:
+    call.inputs.uints[14], //extra:
+    call.inputs.uints[15], //listingTime:
+    call.inputs.uints[16], //expirationTIme:
+    call.inputs.uints[17] //salt:
+  );
+
+  log.info("txHash {} sellOrder maker {} taker {} makerProtocolFee {} takerProtocolFee {} makerRelayerFee {} takerRelayerFee {} feeRecipient {} feeMethod {} side {} saleKind {} exchange {} target{}", 
+    [
+      txHash.toHexString(),
+      sellOrder.maker,
+      sellOrder.taker,
+      sellOrder.makerProtocolFee.toString(),
+      sellOrder.takerProtocolFee.toString(),
+      sellOrder.makerRelayerFee.toString(),
+      sellOrder.takerRelayerFee.toString(),
+      sellOrder.feeRecipient,
+      sellOrder.feeMethod,
+      sellOrder.side,
+      sellOrder.saleKind,
+      sellOrder.exchange.toHexString(),
+      sellOrder.target
+    ]
+  )
+
+  let matchPrice = orders.helpers.calculateMatchPrice(
+    buyOrder,
+    sellOrder,
+    timestamp
+  );
+
+  let allSales = new Array<airstack.nft.Sale>();
+
+  if (isBundleSale) {
+    // log.info("txHash {} bundle sale", [txHash.toHexString()]);
+    let decoded = abi.decodeBatchNftData(
+      buyOrder.callData!,
+      sellOrder.callData!,
+      buyOrder.replacementPattern!
+    );
+
+    for (let i = 0; i < decoded.transfers.length; i++) {
+      log.info("txHash {} transfer method {}", [txHash.toHexString(), decoded.transfers[i].method]);
+      let nft = new airstack.nft.NFT(decoded.addressList[i], decoded.transfers[i].method, decoded.transfers[i].token, decoded.transfers[i].amount);
+      let feeAmount = BigInt.zero();
+      let feeRecipient = "";
+      let creatorRoyaltyFeePercentage = BigInt.zero();
+      let totalRevenueETH = BigInt.zero();
+      let marketplaceRevenueETH = BigInt.zero();
+      let creatorRevenueETH = BigInt.zero();
+      if (sellOrder.feeRecipient!=Address.zero().toHexString()) {
+        creatorRoyaltyFeePercentage = EXCHANGE_MARKETPLACE_FEE.le(
+          sellOrder.makerRelayerFee
+        )
+          ? sellOrder.makerRelayerFee
```

Comment:
Please extract the functions for common logic. 
If & else has same logic except variable change, it code duplication.

Reply:
Done, created a function for it